apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ApmServerConfigurationSource.java

```diff
@@ -202,16 +206,9 @@ private String tryFetchConfig(ConfigurationRegistry configurationRegistry, HttpU
                 logger.debug("Configuration did not change");
                 break;
             case SC_NOT_FOUND:
-                etag = null;
                 // means that there either is no configuration for this agent
                 // or that this is an APM Server < 7.3 which does not have the config endpoint
-                logger.debug("No remote config found for this agent");
-                // makes sure to remove the configuration if all configs are deleted for this agent in Kibana
-                if (!config.isEmpty()) {
-                    logger.info("Received new configuration from APM Server: <empty>");
-                }
-                config = Collections.emptyMap();
-                configurationRegistry.reloadDynamicConfigurationOptions();
```

Comment:
Does this means a user can't cancel configuration (ie restore defaults) by just removing it, and is instead required to actively set everything to specified values?

Reply:
Not sure I completely understand. The user can delete the central configuration in Kibana. The APM Server would then respond with a status code 200 and a `{}` body. That basically sets the `config` map to an empty map.

Reply:
The comment says:
> // means that there either is no configuration for this agent or...

Sorry, I don't remember the spec, but I read that as getting 404 in such case

Reply:
The spec has recently been changed. The APM Server will now only return with 404 if it's pre 7.3: https://github.com/elastic/apm/pull/118

Reply:
removed the outdated comment